cpp/ql/src/Likely Bugs/Underspecified Functions/ArgumentsToImplicit.ql

```diff
@@ -0,0 +1,47 @@
+/**
+ * @name Call with arguments to an implicitly declared function
+ * @description A function call passed arguments even though the
+ *              function in question is only implicitly declared (and
+ *              hence accepting no arguments).  This may indicate
```

Comment:
I thought that we agreed that implicitly declared and <code>()</code>-declared functions should be treated similarly.  Please elaborate.

Reply:
I started to break off `ArgumentsToImplicit.ql`, but then realized that doing so would break our test case.  I believe that `*.qlref` files residing in the same folder should be tested together, and that there is no way for the tests to be separated in some way.  Am I wrong?

I think it would be simpler if  `ArgumentsToImplicit.ql` were reviewed with the other queries.

Reply:
So I think the issue here is quite simple:  If we agree that passing arguments to implicitly-declared functions is something to inform the user about, then we keep the `ArgumentsToImplicit.ql` query.  Otherwise, we simply get rid of it.

Reply:
Calling an implicitly-declared function is dangerous, but I disagree that passing an argument makes it any more dangerous. The compiler will have already warned about the implicit declaration, so I see no urgency to add a query for it.

I don't understand the problem with `qlref` files you describe. With `qltest`, all source files (`c`, `cpp`) in a directory get extracted and imported into a single database, and then each `ql` or `qlref` file is evaluated on that database. These evaluations are independent and produce separate `expected` files.

Reply:
Sorry, got confused by the fact that each `.expected` file contains results for all C/C++ files in the folder.

If you don't believe we need to warn about parameters being passed to implicitly-declared function, then I'll simply remove `ArgumentsToImplicit.ql`.